tests/ui/indexing_slicing_index.stderr

```diff
@@ -1,3 +1,22 @@
+error: indexing may panic
+  --> $DIR/indexing_slicing_index.rs:9:20
+   |
+LL | const REF: &i32 = &ARR[idx()]; // Ok, should not produce stderr.
```

Comment:
These new messages are a result of the change, as the default for `suppress-restriction-lint-in-const` is false. I believe this default makes sense. However, then these comments should be adjusted. Ideally, they should say that is should be linted, since the config value is set.